File: Programming-Basics-With-JavaScript/Homeworks-And-Labs/04.For-Loop/03-additional/03.logistics.js
function logistics(input) {

    const minibusPricePerTon = 200;     // • До 3 тона – микробус (200 лева на тон)
    const truckPricePerTon = 175;       // • От 4 до 11 тона – камион (175 лева на тон)
    const tarinPricePerTon = 120;       // • 12 и повече тона – влак (120 лева на тон)

    const goodsCount = Number(input[0]);
    let index = 1;

    let minibusTonnageCounter = 0;
    let truckTonnageCounter = 0;
    let trainTonnageCounter = 0;
    let totalPrice = 0;

    for (let i = 1; i <= goodsCount; i++) {
        let currentTonnage = Number(input[index++]);
        if (currentTonnage > 11) {
            trainTonnageCounter += currentTonnage;
            totalPrice += currentTonnage * tarinPricePerTon;
        } else if (currentTonnage > 3) {
            truckTonnageCounter += currentTonnage;
            totalPrice += currentTonnage * truckPricePerTon;
        } else {
            minibusTonnageCounter += currentTonnage;
            totalPrice += currentTonnage * minibusPricePerTon;
        }
    }

    const totalTonnage = minibusTonnageCounter + truckTonnageCounter + trainTonnageCounter;
    const averagePricePerTon = totalPrice / totalTonnage;
    const minibusPercentageTonnage = minibusTonnageCounter / totalTonnage * 100;
    const truckPercentageTonnage = truckTonnageCounter / totalTonnage * 100;
    const trainPercentageTonnage = trainTonnageCounter / totalTonnage * 100;

    console.log(averagePricePerTon.toFixed(2));
    console.log(minibusPercentageTonnage.toFixed(2) + "%");
    console.log(truckPercentageTonnage.toFixed(2) + "%");
    console.log(trainPercentageTonnage.toFixed(2) + "%");
}
logistics(["5", "2", "10", "20", "1","7"]);

File: Python_Basics/Homeworks-And-Labs/04.for_loop/03.additional/03.logistics.py
minibus_price_per_ton = 200;     # • До 3 тона – микробус (200 лева на тон)
truck_price_per_ton = 175;       # • От 4 до 11 тона – камион (175 лева на тон)
train_price_per_ton = 120;       # • 12 и повече тона – влак (120 лева на тон)

goods_count = int(input())

minibus_tonnage_counter = 0
truck_tonnage_counter = 0
train_tonnage_counter = 0
total_price = 0

for i in range(goods_count):
    current_tonnage = int(input())
    if current_tonnage > 11:
        train_tonnage_counter += current_tonnage
        total_price += current_tonnage * train_price_per_ton
    elif current_tonnage > 3:
        truck_tonnage_counter += current_tonnage
        total_price += current_tonnage * truck_price_per_ton
    else:
        minibus_tonnage_counter += current_tonnage
        total_price += current_tonnage * minibus_price_per_ton
        
total_tonnage = minibus_tonnage_counter + truck_tonnage_counter + train_tonnage_counter
average_price_per_ton = total_price / total_tonnage
minibus_percentage_tonnage = minibus_tonnage_counter / total_tonnage * 100
truck_percentage_tonnage = truck_tonnage_counter / total_tonnage * 100
train_percentage_tonnage = train_tonnage_counter / total_tonnage * 100

print(f'{average_price_per_ton:.2f}')
print(f'{minibus_percentage_tonnage:.2f}%')
print(f'{truck_percentage_tonnage:.2f}%')
print(f'{train_percentage_tonnage:.2f}%')
